src/server/api/AdminPanel.ts

```diff
@@ -0,0 +1,131 @@
+import type { CustomRequest } from "../../types.js";
+import type { Response } from "express";
+// @ts-ignore
+import { deleteUser, getMemberDataByCriteria, updateMemberColumns } from "../database/memberManager.js";
+// @ts-ignore
+import { deleteRefreshTokensOfUser } from "../database/refreshTokenManager.js";
+// @ts-ignore
+import { logEvents } from "../middleware/logEvents.js";
+
+async function processCommand(req: CustomRequest, res: Response): Promise<void> {
+	const command = req.params["command"]!;
+	const commandAndArgs = command.split(" ");
+	if (!req.memberInfo.signedIn) {
+		res.status(401).send("Cannot send commands while logged out.");
+		return;
+	}
+	if (!(req.memberInfo.roles?.includes("admin") ?? false)) {
+		res.status(403).send("Cannot send commands without the admin role");
+		return;
+	}
+	// TODO prevent affecting accounts with equal or higher roles
+	switch (commandAndArgs[0]) {
+		case "ban":
+			return;
+		case "delete":
+			if (commandAndArgs.length < 2) {
+				res.status(400).send("Invalid number of arguments, expected 1, got " + (commandAndArgs.length - 1) + ".");
+				return;
+			}
+			deleteUser(getMemberDataByCriteria(["user_id"], "username", commandAndArgs[1])["user_id"], commandAndArgs[2] ?? "");
+			res.status(200).send("");
+			logEvents("Command executed: " + command + "\n", "adminCommands");
+			return;
+		case "username":
+			if (commandAndArgs[1] === "get") {
+				if (commandAndArgs.length < 3) {
+					res.status(400).send("Invalid number of arguments, expected 2, got " + (commandAndArgs.length - 1) + ".");
+					return;
+				}
+				res.status(200).send(getMemberDataByCriteria(["username"], "user_id", Number.parseInt(commandAndArgs[2]!))["username"]);
+				logEvents("Command executed: " + command + "\nResult: " + getMemberDataByCriteria(["username"], "user_id", Number.parseInt(commandAndArgs[2]!))["username"] + "\n", "adminCommands");
```

Comment:
These 2 lines:

If getMemberDataByCriteria() will return an empty object if the user is not found. If that happens it looks like it just returns that empty object to the user, which is fine.. but not sure if that's the behavior you wanted.

Also it should be called with the `skipErrorLogging` parameter true so that that function itself doesn't log an error if they don't exist.

"Number.parseInt(commandAndArgs[2]!)" is hard to read when it's inline inside the function. Can you pull it out into its own variable?